Added more double jump enemy bounces to hard logic

LumineonRL opened the pull request:
# Change 1: D03Z03S08 (Grievance Ascends)
Two locations in this room updated to allow for double jump + enemy bounce + (wheel or stall) to cross the gap.

https://discord.com/channels/731205301247803413/1090815418731597884/1251287787198480455


# Change 2: D03Z03S04 (Archcathedral Rooftops)
The health door (D06Z01S04[Health]) can be reached with just double jumps + enemy bounces. Only 1 tick of poison damage is taken with this method and `canSurvivePoison` isn't strictly *necessary*, but I put it on `canSurvivePoison1` just to make it a little easier. I could also see an argument for putting this on `canSurvivePoison2` just to be consistant with everything else in the room.

D06Z01S04[NW] and D06Z01S04[NE] doors did not need to be updated as they already have the necessary double jump + bounce + poison logic from the health door.
Going from the health door to the switch requires several ticks of poison damage, so leaving that on `canSurvivePoison2`

https://discord.com/channels/731205301247803413/1090815418731597884/1250173938516889660
https://discord.com/channels/731205301247803413/1090815418731597884/1250172916126187681
https://discord.com/channels/731205301247803413/1090815418731597884/1250165629470638171


# Change 3:  D06Z01S08 (Archcathedral Rooftops)
Two doors adjusted.
The east entrance can be reached from the west entrance with double jump + enemy bounce or wall climb
The north entrance can be reached from the west or east entrances with double jump + enemy bounce to break the elevator switch and wall climb to reach the door.

https://discord.com/channels/731205301247803413/1090815418731597884/1250201875727519926


# Change 4: D06Z01S15 (Archcathedral Rooftops)

Logic for hitting the switch to access D06Z01S15[NW] and D06Z01S15[NE] updated to include either double jump + enemy bounce or double jump + blood
Logic for collecting the item updated to allow for double jump + enemy bounce

https://discord.com/channels/731205301247803413/1090815418731597884/1250185267709476864
https://discord.com/channels/731205301247803413/1090815418731597884/1250201875727519926
No video for double jump + blood to reach the switch, but it's trivial.

BrandenEK commented on resources/data/Randomizer/doors.json:
```diff
@@ -3523,13 +3523,14 @@
 		"Direction": 0,
 		"OriginalDoor": "D06Z01S13[S]",
-		"VisibilityFlags": "ThisDoor"
+		"VisibilityFlags": "ThisDoor",
```
```suggestion
		"VisibilityFlags": "ThisDoor, EnemySkipsAndDoubleJump",
```
Since there is now a way that this door can be reached other than by itself, we need to add a new visibility flag since it can now also be reached if enemy skips are allowed and double jump is shuffled.  It still can't normally be reached in vanilla though.

BrandenEK commented on resources/data/Randomizer/doors.json:
```diff
@@ -3523,13 +3523,14 @@
 		"Direction": 2,
 		"OriginalDoor": "D06Z01S14[W]",
-		"Logic": "D06Z01S08[N] || D06Z01S08[E] || wallClimb"
+		"Logic": "D06Z01S08[N] || D06Z01S08[E] || wallClimb || doubleJump && canEnemyBounce"
```
These ones that require you to repeatedly spam jump attack should probably have 'preciseSkipsAllowed' as well.  I know it is for this one, but not sure which other ones as well, can you add that where is belongs?

LumineonRL replied:
Agreed. Added `preciseSkippsAllowed` to D06Z01S04[Health] and D06Z01S08[N].

BrandenEK commented:
For future reference,

canSurvivePoison1: Small enough that is can be completely ran through with only basic health.  However, will require tiento on lower difficulties.
canSurvivePoison2: Can not be ran through and requires tiento on all difficulties.  This includes climbing sections in poison.
canSurvivePoison3: Requires multiple casts of tiento.  This really only applies to those 2 long hallways.

LumineonRL commented:
Updated PR to add `preciseSkipsAllowed` where multiple double jump + bounces are required, and removed redundant `canSurvivePoison1` requirement for reaching the `D06Z01S04[Health]` door.
